core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -172,13 +187,26 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
     trace(s"Added marker ${txnIdAndMarker.txnMarkerEntry} for transactional id ${txnIdAndMarker.txnId} to destination broker $brokerId")
   }
 
+  def retryLogAppends(): Unit = {
+    val txnLogAppendRetries: java.util.List[TxnLogAppend] = new util.ArrayList[TxnLogAppend]()
+    txnLogAppendRetryQueue.drainTo(txnLogAppendRetries)
+    debug(s"retrying: ${txnLogAppendRetries.size} transaction log appends")
+    txnLogAppendRetries.asScala.foreach { txnLogAppend =>
+      txnLogAppendRetryQueueSensor.record(-1)
```

Comment:
We could just call txnLogAppendRetryQueueSensor.record(txnLogAppendRetries.size()) outside the loop?